tests/gocase/unit/config/config_test.go

```diff
@@ -266,6 +266,12 @@ func TestChangeProtoMaxBulkLen(t *testing.T) {
 	require.NoError(t, err)
 	require.EqualValues(t, "2097152", vals["proto-max-bulk-len"])
 
+	// change to 100MB
+	require.NoError(t, rdb.ConfigSet(ctx, "proto-max-bulk-len", "100M").Err())
+	vals, err = rdb.ConfigGet(ctx, "proto-max-bulk-len").Result()
+	require.NoError(t, err)
+	require.EqualValues(t, "100M", vals["proto-max-bulk-len"])
```

Comment:
For the config get command, Redis always returns an integer instead of the value with unit, for the sake of many monitoring tools will use this value as an integer.

Reply:
I don't think there can be 2 different `ToString()` for two purpose. As it can make the config part hard to maintain.

Reply:
also I think we should keep the original format instead of convert the value in `CONFIG GET`.

Reply:
Yes, I agree we should not have two `ToString()` for this purpose. But perhaps we can convert the value to string this in Gonfig::Get if it's an integer field. What do you think?

Reply:
>  Redis always returns an integer instead of the value with unit

Redis doesn't support such config format (integer with unit) at all, so I don't think we should align with Redis.

> convert the value to string this in Gonfig::Get if it's an integer field

Sorry I cannot get your point. Could you explain it?

Reply:
> Redis doesn't support such config format (integer with unit) at all, so I don't think we should align with Redis.


https://github.com/redis/redis/blob/unstable/redis.conf#L2111 Redis also supports the value with value.

> Sorry I cannot get your point. Could you explain it?

My initial thought is to use `std::string()` instead of `ToString()` if the field type is InterFieldWithUnit.

Reply:
> My initial thought is to use std::string() instead of ToString() if the field type is InterFieldWithUnit.

IMHO it's even worse than two virtual function inside ConfigField in the aspect of software engineering...

Reply:
I see, might we also add another virtual function for the config dump purpose.